src/middlewared/middlewared/pytest/unit/plugins/test_smart.py

```diff
@@ -1,23 +1,52 @@
 import textwrap
+import json
 
 import pytest
 
 from middlewared.plugins.smart import parse_smart_selftest_results, parse_current_smart_selftest
 
 
 def test__parse_smart_selftest_results__ataprint__1():
-    assert parse_smart_selftest_results(textwrap.dedent("""\
-        smartctl 6.6 2017-11-05 r4594 [FreeBSD 11.1-STABLE amd64] (local build)
-        Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org
-
-        === START OF READ SMART DATA SECTION ===
-        SMART Self-test log structure revision number 1
-        Num  Test_Description    Status                  Remaining  LifeTime(hours)  LBA_of_first_error
-        # 1  Short offline       Completed without error       00%     16590         -
-        # 2  Short offline       Completed without error       00%     16589         -
-    """)) == [
+    data = json.loads(textwrap.dedent("""\
+{
+    "ata_smart_self_test_log": {
+        "standard": {
+            "revision": 1,
+            "table": [
+                {
+                    "type": {
+                        "value": 1,
+                        "string": "Short offline"
+                    },
+                    "status": {
+                        "value": 0,
+                        "string": "Completed without error",
+                        "passed": true
+                    },
+                    "lifetime_hours": 16590
+                },
+                {
+                    "type": {
+                        "value": 1,
+                        "string": "Short offline"
+                    },
+                    "status": {
+                        "value": 0,
+                        "string": "Completed without error",
+                        "passed": true
+                    },
+                    "lifetime_hours": 16589
+                }
+            ],
+            "error_count_total": 0,
+            "error_count_outdated": 0
+            }
+        }
+}
+        """))
+    assert parse_smart_selftest_results(data) == [
         {
-            "num": 1,
+            "num": 0,
```

Comment:
Why do results change?

Reply:
Between ata, nvme, and iscsi, there are inconsistencies between the test number being 0 indexed (or included at all in the JSON). Sometimes even in the same drive type (iirc iscsi's normal output and JSON output are 0 and 1 indexed respectively lol).
This was to standardize it all to zero indexed